frontend/lib/resolution/scope-queries.cpp

```diff
@@ -616,6 +616,27 @@ getKindForVisibilityClauseId(Context* context, ID visibilityClauseId) {
   return VIS_USE;
 }
 
+// Creates a BorrowedIdsWithName for the symbol that defines a scope.
+// Used for capturing 'IO' when a visibility statement is 'use IO'.
+static optional<BorrowedIdsWithName>
+borrowedIdsForScopeSymbol(Context* context,
+                          IdAndFlags::Flags filterFlags,
+                          const IdAndFlags::FlagSet& excludeFilter,
+                          const Scope* scope) {
+  auto scopeAst = parsing::idToAst(context, scope->id());
+  auto visibility = scopeAst->toDecl()->visibility();
```

Comment:
I know you are just moving this code, but could you fix this issue with it?

It should use `parsing::idIsPrivateDecl` to avoid calling `idToAst` which is both slow and unlikely to be reusable from revision to revision

Reply:
That doesn't quite work because the argument expects a `Visibility`, not a boolean. But I'll add a new parsing query to retrieve the visibility.

Reply:
AFAIK, we don't need to distinguish here between default visibility and public. You could just write

```
auto visibility = idIsPrivateDecl(context, scope->id()) ? Decl::PRIVATE : Decl::Public
```

Distinguishing between these two is useful in the uAST but I can't imagine any reason it would matter in scope resolution (and in fact, if it does matter, it's probably a bug, because the default visibility is public).

Of course you could put this in a helper in parsing-queries.